File: csharp/markdown/Markdown.cs
using System.Text.RegularExpressions;
public static class Markdown
{
    public static string Parse(string input)
    {
        var lvl = Regex.Match(input, "(?m)^(#{1,6})").Groups[1].Length;
        input = Regex.Replace(input, "(?m)^#{1,6} (.+)$", $"<h{lvl}>$1</h{lvl}>");
        input = Regex.Replace(input, "__(.+)__", "<strong>$1</strong>");
        input = Regex.Replace(input, "_(.+)_", "<em>$1</em>");
        input = Regex.Replace(input, @"(?m)^\* (.+)$", "<li>$1</li>");
        input = Regex.Replace(input, "(?s)(<li>.*</li>)", "<ul>$1</ul>");
        input = Regex.Replace(input, "(?m)^(?!<h|<l|<u)(.+)$", "<p>$1</p>");
        return  Regex.Replace(input, "\n", "");
    }
}

File: fsharp/markdown/Markdown.fs
module Markdown
open System.Text.RegularExpressions
let parse md =
   let lvl  = Regex.Match(md, "(?m)^(#{1,6})").Groups[1].Length
   let mutable html = Regex.Replace(md, "(?m)^#{1,6} (.+)$", $"<h{lvl}>$1</h{lvl}>")
   html <- Regex.Replace(html, "__(.+)__", "<strong>$1</strong>")
   html <- Regex.Replace(html, "_(.+)_", "<em>$1</em>")
   html <- Regex.Replace(html, "(?m)^\* (.+)$", "<li>$1</li>")
   html <- Regex.Replace(html, "(?s)(<li>.*</li>)", "<ul>$1</ul>")
   html <- Regex.Replace(html, "(?m)^(?!<h|<l|<u)(.+)$", "<p>$1</p>")
   Regex.Replace(html, "\n", "")